DESIGN-PROPOSALS.md

```diff
@@ -0,0 +1,93 @@
+# Design Proposal Template
+
+**Authors**: alice alice@example.com, bob jones bob@example.com
+
+**Begin Design Discussion**: YYYY-MM-DD
+
+**(optional) Status: **implementable, accepted, deferred, rejected, withdrawn, or replaced
+
+**Checklist**:
+
+- [ ] Prerequisite 1
+- [ ] Prerequisite 2
+- [ ] Component 1
+- [ ] Component 2
+- [ ] Docs
+- [ ] Tests
```

Comment:
```suggestion
- [ ] Problem, need, impact, and solution definition
- [ ] User Story/Stories
- [ ] Proposal scoping (goals/non-goals)
- [ ] Detailed design, diagrams
- [ ] Impact, risk, and security considerations
- [ ] Observability, performance, and scaling considerations
- [ ] Availability, compatibility, and roll back considerations
- [ ] Implementation phases, planning timeline
- [ ] Test plan
- [ ] Accept/reject proposal
- [ ] Implementation
- [ ] Docs updates and release notes
- [ ] Release
```

I've suggested a more specific listing that aligns with the content of this doc. If this is more what you are looking for, i would recommend shifting the ordering of the doc to reflect the list here so authors can follow along (it'd be sweet if these were links to the sections of the doc, then it can be used as a Table of Contents )

Reply:
Well, I was mostly trying to indicate that there should be some kind of checklist there, but what goes *in* the checklist is highly dependent on the project's own processes.  For example, KEPs have a very different-looking checklist: https://github.com/kubernetes/enhancements/issues/3668 .  I was trying NOT to have anything that suggested a specific process that probably wouldn't apply.

Reply:
I agree with Josh that this should be a checklist and not a table of contents.

Reply:
Sounds good.  Providing instructions for this checklist would be helpful, particularly describing what it is intended to capture.